tools/activate_intel_oneAPI_INCLUDE.py

```diff
@@ -0,0 +1,24 @@
+import os
+
+msvc_path = ("C:\\Program Files\\Microsoft Visual Studio\\2022\\"
+             "Enterprise\\VC\\Tools\\MSVC\\")
+version_dir = os.listdir(msvc_path)
+vc_version = version_dir[-1]
+
+INCLUDE = ("C:\\Program Files\\Microsoft Visual Studio\\2022\\"
+           fr"Enterprise\VC\Tools\MSVC\{vc_version}\include;"
+           "C:\\Program Files\\Microsoft Visual Studio\\2022\\"
+           fr"Enterprise\VC\Tools\MSVC\{vc_version}\ATLMFC\include;"
+           "C:\\Program Files\\Microsoft Visual Studio\\2022\\"
+           r"Enterprise\VC\Auxiliary\VS\include;"
+           r"C:\Program Files (x86)\Windows Kits\10\include\10.0.22621.0\ucrt;"
```

Comment:
I'm not quite understanding these scripts. @h-vetinari asked to do this programmatically rather than hardcode the paths with specific version numbers. It's moved to a `.py` file now, but it's still hardcoded.

Reply:
I am able to detect MSVC versions and hence those are added according to the ones present on the system. However these are Windows OS related versions. I tried detecting those as well. On my Azure machine it works, but here on CI the they are always empty. Calling batch files for MSVC and Intel oneAPI is not an option on CI because of https://github.com/scipy/scipy/pull/20985#issuecomment-2190002406.

The general structure of the paths remain the same, only the version number in those paths change. I think setting OS to windows-2022 should keep the windows related version numbers stable. For MSVC I already handle those pragmatically.

Reply:
I guess I found something relevant, https://github.com/conan-io/cmake-conan/issues/465#issuecomment-1336335529.

Reply:
I will try this on my personal branch and update here if it works.